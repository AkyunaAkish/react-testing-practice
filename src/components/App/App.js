import React, { PureComponent } from 'react';
import { Button } from 'react-bootstrap';

class App extends PureComponent {
    state = {
        gifts: []    
    }

    addGift = () => {
        const { gifts } = this.state;

        const ids = this.state.gifts.map((gift => gift.id));
        
        // spread id array as args and get highest number in ids arr
        const maxId = ids.length > 0 ? Math.max(...ids) : 0; 

        gifts.push({ id: maxId+1 });

        // update gifts with new modified copy
        this.setState({ gifts });
    }

    render() {
        return (
            <div className='app'>
                <h1>Gift Giver</h1>
                <Button className='btn-add'
                        onClick={ this.addGift }>
                    Add Gift
                </Button>
            </div>
        );
    }
}

export default App;