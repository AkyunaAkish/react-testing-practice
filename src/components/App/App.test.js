import React from 'react';
import { shallow } from 'enzyme';

import App from './App';
const app = shallow(<App />);

// Whenever you make a change to the app component
// that you want to keep, go to the new failing test
// and press "U" in the command line to update your snapshoit

it('renders <App /> correctly', () => {
    // snapshots keep a recorded history of 
    // your react component
    // whenever a change happens in the app
    // snapshots will update and keep a history
    // of the rendering of the component
    expect(app).toMatchSnapshot();
});

it('initializes <App /> `state` with an empty list of gifts', () => {
    expect(app.state().gifts).toEqual([]);
});

it('adds a new gift to  <App /> `state` when clicking `add gift` button', () => {
    app.find('.btn-add').simulate('click');

    expect(app.state().gifts).toEqual([{ id: 1 }]);
});

