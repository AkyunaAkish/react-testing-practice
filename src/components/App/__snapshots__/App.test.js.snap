// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <App /> correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Gift Giver
        </h1>,
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="default"
          className="btn-add"
          disabled={false}
          onClick={[Function]}
        >
          Add Gift
        </Button>,
      ],
      "className": "app",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Gift Giver",
        },
        "ref": null,
        "rendered": "Gift Giver",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "active": false,
          "block": false,
          "bsClass": "btn",
          "bsStyle": "default",
          "children": "Add Gift",
          "className": "btn-add",
          "disabled": false,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Add Gift",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Gift Giver
          </h1>,
          <Button
            active={false}
            block={false}
            bsClass="btn"
            bsStyle="default"
            className="btn-add"
            disabled={false}
            onClick={[Function]}
          >
            Add Gift
          </Button>,
        ],
        "className": "app",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Gift Giver",
          },
          "ref": null,
          "rendered": "Gift Giver",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "active": false,
            "block": false,
            "bsClass": "btn",
            "bsStyle": "default",
            "children": "Add Gift",
            "className": "btn-add",
            "disabled": false,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Add Gift",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
